client/app/services/event-notifications.service.js

```diff
@@ -42,21 +42,73 @@
       }
     };
 
+    function miqFormatNotification(text, bindings) {
+      var str = __(text);
+      lodash.each(bindings, function (value, key) {
+        str = str.replace(new RegExp('%{' + key + '}', 'g'), value.text);
+      });
+
+      return str;
+    }
+
     doReset();
 
+    var cable = ActionCable.createConsumer('/ws/notifications');
+
+    cable.subscriptions.create('NotificationChannel', {
+      disconnected: function () {
+        // eslint-disable-next-line consistent-this
```

Comment:
Consistency good, though if feels weird outside controllers :) (As in, every time I see it, I think I'm in a controller.) (Also in the context of manageiq, `vm` is a bit unfortunate.. as in, at some point, we'll have a VmController to show the list of vms, and we'll be ..uh, hunter2ed :).)